packages/lit-dev-content/site/blog/2022-07-19-lit-for-polymer-users.md

```diff
@@ -0,0 +1,935 @@
+
+_setName(name) {
+  this._name = name;
+  this.requestUpdate('name');
+}
+```
+
+{% endswitchable-sample %}
+
+Note that if the property isn't included in the component's template, you don't need to declare it  (with `@property` or `static properties`) or call `requestUpdate()`.
+
+
+### Computed properties and observers
+
+Lit provides a number of overridable lifecycle methods that are called when reactive properties change. Use these methods to implement logic that would go in computed properties or observers in Polymer.
+
+The following lists summarize how to migrate different kinds of computed properties and observers.
+
+Computed properties:
+
+*   For values used ephemerally in the template, compute values as local variables in `render()` and use them in the template before returning.
+
+*   For values that need to be stored persistently, or are expensive to compute, do so in `willUpdate()`.
+
+*   Use the `changedProperties.has()` method to compute only when dependencies change, to avoid expensive re-computation every update.
+
+Observers:
+
+*   If the observer needs to act directly on DOM based on property changes, use `updated()`. This is called after the `render()` callback.
+
+*   Otherwise, use `willUpdate()`.
+
+*   In either case, use `changedProperties.has()` to know when a property has changed.
+
+Path-based observers:
+
+*   These are complex observers that observe paths like `foo.bar` or `foo.*`:
+
+    ```js
+    observers: ['fooBarChanged(foo.bar)', 'fooChanged(foo.*)']
+    ```
+
+*   This feature was very specific to Polymer's data system that has no equivalent in Lit.
+
+*   We recommend using immutable patterns as a more interoperable way to communicate deep property changes. For more information, see [Mutating object and array properties](https://lit.dev/docs/components/properties/#mutating-properties).
+
+
+#### Computing transient values in render()
+
+If you need to calculate transient values that are only used for rendering, you can calculate them directly in the `render()` method.
+
+Polymer:
+
+```js
+ static get template() {
+    return html`
+      <p>Text: <span>{{cyphertext}}</span></p>
+    `;
+  }
+
+  static get properties() {
+    return {
+      /* ... */
+      cyphertext: {
+        type: String,
+        computed: 'rot13(cleartext)'
+      }
+    }
+  }
+```
+
+Lit:
+
+```js
+render() {
+  const cyphertext = rot13(this.cleartext);
+  return html`Text: ${cyphertext}`;
+}
+```
+
+Also, because Lit allows you to use any JavaScript expression in a template, Polymer's [computed bindings](https://polymer-library.polymer-project.org/3.0/docs/devguide/data-binding#annotated-computed) can be replaced with inline expressions, including function calls.
+
+Polymer:
+
+```js
+  static get template() {
+    return html`
+      My name is <span>[[_formatName(given, family)]]</span>
+    `;
+  }
+
+```
+
+Lit:
+
+```js
+render() {
+  html`
+      My name is <span>${this._formatName(given, family)}</span>
```

Comment:
Then this is sooo much better.

```suggestion
      My name is <span>${this.given} ${this.family}</span>
```

Reply:
In this case both Polymer and Lit could simply insert two strings. Calling a function is more interesting because, for example, formatting the name might be locale-specific. And it's a very easy pattern to translate.

Reply:
I'm not making these two changes because this section is specifically showing how to use an inline function.